src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeHelpersTests.cs

```diff
@@ -193,16 +192,57 @@ private static void FillStack(int depth)
             }
         }
 
-        [Fact]
-        public static void GetUninitializedObject_InvalidArguments_ThrowsException()
+        public static IEnumerable<object[]> GetUninitializedObject_NegativeTestCases()
+        {
+            // TODO: Test actual function pointer types when typeof(delegate*<...>) support is available
+
+            Type canonType = typeof(object).Assembly.GetType("System.__Canon", throwOnError: true);
+
+            yield return new[] { typeof(string), typeof(ArgumentException) }; // variable-length type
+            yield return new[] { typeof(int[]), typeof(ArgumentException) }; // variable-length type
+            yield return new[] { typeof(int[,]), typeof(ArgumentException) }; // variable-length type
+            yield return new[] { Array.CreateInstance(typeof(int), new[] { 1 }, new [] { 1 }).GetType(), typeof(ArgumentException) }; // variable-length type (non-szarray)
+            yield return new[] { typeof(Array), typeof(MemberAccessException) }; // abstract type
+            yield return new[] { typeof(Enum), typeof(MemberAccessException) }; // abstract type
+
+            yield return new[] { typeof(Stream), typeof(MemberAccessException) }; // abstract type
+            yield return new[] { typeof(IDisposable), typeof(MemberAccessException) }; // interface type
+
+            yield return new[] { typeof(List<>), typeof(MemberAccessException) }; // open generic type
+            yield return new[] { typeof(List<>).GetGenericArguments()[0], typeof(ArgumentException) }; // 'T' placeholder typedesc
+
+            yield return new[] { typeof(Delegate), typeof(MemberAccessException) }; // abstract type
+            yield return new[] { typeof(Action), typeof(ArgumentException) }; // delegate type
+
+            yield return new[] { typeof(void), typeof(ArgumentException) }; // explicit block in place
+            yield return new[] { typeof(int).MakePointerType(), typeof(ArgumentException) }; // pointer typedesc
+            yield return new[] { typeof(int).MakeByRefType(), typeof(ArgumentException) }; // byref typedesc
+
+            yield return new[] { typeof(ReadOnlySpan<int>), typeof(NotSupportedException) }; // byref type
+            yield return new[] { typeof(ArgIterator), typeof(NotSupportedException) }; // byref type
+            yield return new[] { typeof(List<>).MakeGenericType(canonType), typeof(NotSupportedException) }; // shared by generic instantiations
+
+            if (PlatformDetection.IsWindows)
```

Comment:
This should not be IsWindows but IsComSupported

Reply:
`PlatformDetection.IsComSupported` is https://github.com/dotnet/runtime/issues/34360 that was not fixed yet

Reply:
Looks like that test API is not available yet but is being tracked by https://github.com/dotnet/runtime/issues/35079. Would it make sense to leave this as-is but put a comment linking to that issue?

Reply:
Another option is a quick n' dirty implementation:

```cs
public static bool IsComSupported => IsCoreClr && IsWindows;
```

It matches the ifdefs already present in the coreclr project.

https://github.com/dotnet/runtime/blob/464508b9d15371b1a972de34b428b3e9ed6181b1/src/coreclr/clr.featuredefines.props#L26-L38

Reply:
quick n' dirty implementation is better than a comment

Reply:
Maybe submit the tests in separate PR. It will also make it easy to see that they pass both before and after the change.

Reply:
Latest iteration introduced `PlatformDetection.SupportsComInterop` (to match other `Supports*` APIs already on that type). I'm leaving the thread unresolved, though, to track splitting the tests into a different PR entirely.

Reply:
New PR is open specifically to track the unit test issue, resolving this thread.